Content.Server/Radio/EntitySystems/JammerSystem.cs

```diff
@@ -78,11 +84,47 @@ private void OnRadioSendAttempt(ref RadioSendAttemptEvent args)
         var query = EntityQueryEnumerator<ActiveRadioJammerComponent, RadioJammerComponent, TransformComponent>();
         while (query.MoveNext(out _, out _, out var jam, out var transform))
         {
-            if (source.InRange(EntityManager, _transform, transform.Coordinates, jam.Range))
+            if (source.InRange(EntityManager, _transform, transform.Coordinates, GetCurrentRange(jam)))
             {
                 args.Cancelled = true;
                 return;
             }
         }
     }
+
+    private void OnGetVerb(Entity<RadioJammerComponent> entity, ref GetVerbsEvent<Verb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract)
+            return;
+
+        var user = args.User;
+
+        for (byte i = 0; i < entity.Comp.Settings.Count; i++)
+        {
+            var currentIndex = i;
+            var verb = new Verb
+            {
+                Priority = currentIndex,
+                Category = VerbCategory.PowerLevel,
+                Disabled = entity.Comp.SelectedPowerLevel == currentIndex,
+                Act = () =>
+                {
+                    entity.Comp.SelectedPowerLevel = currentIndex;
+                    _popup.PopupEntity(Loc.GetString(entity.Comp.Settings[currentIndex].Message), user, user);
+                },
+                Text = Loc.GetString(entity.Comp.Settings[currentIndex].Name),
+            };
+            args.Verbs.Add(verb);
+        }
+    }
+    private static float GetCurrentWattage(RadioJammerComponent jammer)
+    {
+        var selectedPowerLevel = jammer.SelectedPowerLevel;
```

Comment:
uses more characters than it saves :trollface:

Reply:
Haha yeah that's fair I just did it that way because I thought it was a little clearer what was going on!